File: incident-service/src/main/java/com/emergency/incident/model/Incident.java
package com.emergency.incident.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "incidents")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Incident {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "citizen_name", nullable = false)
    private String citizenName;

    @Column(name = "citizen_phone")
    private String citizenPhone;

    @Enumerated(EnumType.STRING)
    @Column(name = "incident_type", nullable = false)
    private IncidentType incidentType;

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private Double longitude;

    @Column(name = "location_address")
    private String locationAddress;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(name = "created_by", nullable = false)
    private Long createdBy;

    @Column(name = "assigned_unit_id")
    private Long assignedUnitId;

    @Column(name = "assigned_unit_name")
    private String assignedUnitName;

    @Column(name = "assigned_unit_type")
    private String assignedUnitType;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private IncidentStatus status = IncidentStatus.CREATED;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime timestamp;

    @Column(name = "dispatched_at")
    private LocalDateTime dispatchedAt;

    @Column(name = "resolved_at")
    private LocalDateTime resolvedAt;
}

File: backend/incident-service/src/entities/Incident.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  UpdateDateColumn,
} from 'typeorm';

export enum IncidentStatus {
  CREATED = 'CREATED',
  DISPATCHED = 'DISPATCHED',
  IN_PROGRESS = 'IN_PROGRESS',
  RESOLVED = 'RESOLVED',
  CANCELLED = 'CANCELLED',
}

export enum IncidentType {
  FIRE = 'FIRE',
  MEDICAL = 'MEDICAL',
  ACCIDENT = 'ACCIDENT',
  HAZMAT = 'HAZMAT',
  OTHER = 'OTHER',
}

@Entity('incidents')
export class Incident {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column()
  citizenName!: string;

  @Column()
  citizenPhone!: string;

  @Column({ type: 'enum', enum: IncidentType })
  incidentType!: IncidentType;

  @Column('decimal', { precision: 10, scale: 8 })
  latitude!: number;

  @Column('decimal', { precision: 11, scale: 8 })
  longitude!: number;

  @Column()
  locationAddress!: string;

  @Column({ nullable: true })
  notes!: string;

  @Column()
  createdBy!: number;

  @Column({ type: 'enum', enum: IncidentStatus, default: IncidentStatus.CREATED })
  status!: IncidentStatus;

  @Column({ nullable: true })
  assignedUnitId!: number;

  @Column({ nullable: true })
  assignedUnitName!: string;

  @Column({ nullable: true })
  assignedUnitType!: string;

  @CreateDateColumn()
  createdAt!: Date;

  @Column({ nullable: true })
  dispatchedAt!: Date;

  @UpdateDateColumn()
  updatedAt!: Date;
}
